app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -636,24 +637,28 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
                 true
             }
             else -> {
-                // In case we're displaying search results, we wouldn't have navigated to home, and
-                // so we don't need to navigate "back to" browser fragment.
-                // See mirror of this logic in BrowserToolbarController#handleToolbarClick.
-                if (store.state.searchTerms.isBlank()) {
-                    val args by navArgs<SearchDialogFragmentArgs>()
-                    args.sessionId?.let {
-                        findNavController().navigate(
-                            SearchDialogFragmentDirections.actionGlobalBrowser(null),
-                        )
-                    }
-                }
-                view?.hideKeyboard()
-                dismissAllowingStateLoss()
+                goBack()
                 true
             }
         }
     }
 
+    private fun goBack() {
```

Comment:
I like that this has been extracted to a function 👍 
Just looking at the comment

> // See mirror of this logic in BrowserToolbarController#handleToolbarClick.

Is this still relevant, as that logic block doesn't appear to mirror this necessarily?

Reply:
I believe this refers to the case when interacting with the toolbar from a tab without search terms (which does not have search terms set). In `handleToolbarClick` we navigate home before showing the search dialog for the home fragment to be visible in the decor. In `onBackPressed` from `SearchDialogFragment` we want to return to the browser fragment when dismissing the search dialog, instead of the home fragment. It seems to me that the comment is still relevant, having this case in mind. Should we maybe update it to be more descriptive?

Reply:
I see, gotcha, thanks for explaining 👍 I misinterpreted this as literally mirroring the logic. Happy to keep comment as is, unless you have an update in mind?

Reply:
I'm ok with it as is too, thanks!